### PR TITLE
Improve food transports display

### PR DESCRIPTION
**Pull request by magopian**
[user story](https://www.notion.so/TRANSPORTS-Isoler-les-transports-vers-les-sites-de-stockage-et-de-distribution-f39e06680e2749d79778563275cc9d43)

### CODE REVIEW

File: src/Data/Food/Retail.elm [hunk 1]
```diff
@@ -186,24 +185,34 @@ elecImpact elecNeeds volume =
...
-    .impacts
-        >> Impact.mapImpacts
-            (\_ impact ->
-                impact
-                    |> Unit.impactToFloat
-                    |> (*) (Length.inKilometers distance * Mass.inMetricTons mass)
-                    |> Unit.impact
-            )
+transportImpact : Db -> Mass -> Distribution -> WellKnown -> Transport
+transportImpact db mass (Distribution _ needs) wellKnown =
```
Comment (n1k0) [inline src/Data/Food/Retail.elm:189]: Nit: as this now strictly depends on `Distribution`, we should probably rename this function to `distributionTransportImpact`

File: src/Page/Food/Builder.elm [hunk 2]
```diff
@@ -590,50 +591,80 @@ updateIngredientFormView { excluded, db, ingredient, impact, transportImpact } =
...
-            , if ingredient.planeTransport /= Ingredient.PlaneNotApplicable then
-                label
-                    [ class "PlaneCheckbox ps-2" ]
-                    [ text "("
-                    , input
-                        [ type_ "checkbox"
-                        , class "form-check-input no-outline"
+        , displayTransportDistances db ingredient ingredientQuery event
+        , span
+            [ class "text-muted text-end IngredientTransportImpact fs-7"
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:596]: btw, please apply the `fs-7` class to the element where `ImpactDisplay` is used, so we avoid display issue for larger numbers

image
image

File: src/Page/Food/Builder.elm [hunk 3]
```diff
@@ -590,50 +591,80 @@ updateIngredientFormView { excluded, db, ingredient, impact, transportImpact } =
...
+                        , { distance = road, icon = Icon.bus, label = "Transport routier" }
+                        ]
+                            |> List.filterMap
+                                (\{ distance, icon, label } ->
+                                    if Length.inKilometers distance /= 0 then
+                                        Just <| TransportView.entry distance icon label
+
+                                    else
+                                        Nothing
+                                )
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:665]: Nit: Wouldn't this a good use case for adapting/extending `TransportView.viewDetails`?

File: src/Page/Food/Builder.elm [hunk 4]
```diff
@@ -773,11 +804,36 @@ packagingListView db selectedImpact recipe results =
...
 
+transportToDistributionView : Impact.Definition -> Recipe -> Recipe.Results -> List (Html Msg)
+transportToDistributionView selectedImpact recipe results =
+    [ span [ class "w-50" ] [ text "" ]
+    , div [ class "DownArrow" ]
+        [ text "" ]
+    , div
+        [ class "w-50"
+        , style "padding" ".5rem 1rem"
+        ]
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:815]: Please add `fs-7` text style here, for styling consistency

File: src/Page/Food/Builder.elm [hunk 5]
```diff
@@ -773,11 +804,36 @@ packagingListView db selectedImpact recipe results =
     ]
 
 
+transportToDistributionView : Impact.Definition -> Recipe -> Recipe.Results -> List (Html Msg)
+transportToDistributionView selectedImpact recipe results =
+    [ span [ class "w-50" ] [ text "" ]
+    , div [ class "DownArrow" ]
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:810]: Note: 
- it's a little strange we have an arrow here but not for the other steps
- also ideally we'd update other down arrows to use this new style (maybe in a later patch)

File: src/Page/Food/Builder.elm [hunk 6]
```diff
@@ -773,11 +804,36 @@ packagingListView db selectedImpact recipe results =
...
+        , style "padding" ".5rem 1rem"
+        ]
+        [ div []
+            [ text "Masse : "
+            , Recipe.getTransformedIngredientsMass recipe
+                |> Format.kg
+            , text " + Emballage : "
+            , Recipe.getPackagingMass recipe
+                |> Format.kg
+            ]
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:823]: Have you tried moving this line to the left side of the down arrow?

Example:

```elm
transportToDistributionView : Impact.Definition -> Recipe -> Recipe.Results -> List (Html Msg)
transportToDistributionView selectedImpact recipe results =
    [ div [ class "w-50 fs-7 px-2 py-3 text-center" ]
        [ text "Masse : "
        , Recipe.getTransformedIngredientsMass recipe
            |> Format.kg
        , text " + Emballage : "
        , Recipe.getPackagingMass recipe
            |> Format.kg
        ]
    , div [ class "DownArrow" ] []
    , div [ class "w-50 fs-7 px-2 py-3" ]
        [ div [ class "d-flex justify-content-around" ]
            [ TransportView.entry results.distribution.transports.road Icon.bus "Transport routier"
            , Format.formatFoodSelectedImpact selectedImpact results.distribution.transports.impacts
            ]
        ]
    ]
```

image

File: src/Page/Food/Builder.elm [hunk 7]
```diff
@@ -773,11 +804,36 @@ packagingListView db selectedImpact recipe results =
     ]
 
 
+transportToDistributionView : Impact.Definition -> Recipe -> Recipe.Results -> List (Html Msg)
+transportToDistributionView selectedImpact recipe results =
+    [ span [ class "w-50" ] [ text "" ]
+    , div [ class "DownArrow" ]
+        [ text "" ]
```
Comment (n1k0) [inline src/Page/Food/Builder.elm:811]: Nit: This empty text node is not unnecessary

**Review by n1k0 — APPROVED** [review]
Great work, I love the new transport toggler! A few nits and remarks.

File: tests/Data/Food/Builder/RecipeTest.elm [hunk 8]
```diff
@@ -93,7 +93,7 @@ suite =
                             , all = { impact = Unit.impact 169.83232687747997, letter = "B", outOf100 = 72 }
                             , biodiversity = { impact = Unit.impact 131.39490621544522, letter = "B", outOf100 = 62 }
                             , health = { impact = Unit.impact 46.77851948542256, letter = "B", outOf100 = 73 }
-                            , resources = { impact = Unit.impact 23.268322876906748, letter = "B", outOf100 = 74 }
+                            , resources = { impact = Unit.impact 23.26832287690675, letter = "B", outOf100 = 74 }
```
[the remark this answers is not in the document]
Comment (magopian) [reply tests/Data/Food/Builder/RecipeTest.elm:96]: Good idea, I've added a small helper to test scorings.

File: src/Page/Food/Builder.elm (hunk 6 above)
Comment (magopian) [reply src/Page/Food/Builder.elm:823]: It looks nice indeed, however I've implemented as discussed and iterated with Pascal and Laurent (see the notion user story ;)

File: src/Page/Food/Builder.elm (hunk 3 above)
Comment (magopian) [reply src/Page/Food/Builder.elm:665]: good idea ;)

File: src/Page/Food/Builder.elm (hunk 5 above)
Comment (magopian) [reply src/Page/Food/Builder.elm]: Yeah, we only need some information here, but we could still have arrows between each step, I'll add this.

I wasn't sure you wanted me to replace the textile arrows, but I can ;)

**Comment by magopian** [comment]
Ok I believe i've addressed all your feedback, care to give it a quick last look before I merge?

**Comment by n1k0** [comment]
Looks awesome, ship it! (we can always get back to the down arrow texts later)

**Review by n1k0 — APPROVED** [review]

**Pull request merged**